vlog4j-core/src/main/java/org/semanticweb/vlog4j/core/reasoner/Reasoner.java

```diff
@@ -86,6 +87,20 @@ static Reasoner getInstance() {
 	 */
 	KnowledgeBase getKnowledgeBase();
 
+	/**
+	 * Exports the inferred facts of the knowledge base to an OutputStream.
+	 * 
+	 * @param an OutputStream for the facts to be written to.
+	 */
+	void writeInferredFacts(OutputStream stream) throws IOException;
```

Comment:
I thought we want to obtain the whole chase result, not only the inferred facts.
Anyway, the method returns all inferred facts plus some facts that were added to the KB as Fact objects.
If we only want to obtain inferred facts, we should not return explicit facts; if we want to obtain the entire chase result, we should add all explicit facts (i.e. the ones added to the KB from a `DataSource`).

Reply:
@irina-dragoste  I query over all predicates, actually isn't the facts obtained somehow from datasources are either part of the facts of the knowledge base which I obtain or at least they have a predicate which I query for ?, are u sure this is not covered ?

Reply:
@irina-dragoste  nvm done